pkg/api/v1/workloads.go

```diff
@@ -687,6 +690,35 @@ type updateRequest struct {
 	ToolsFilter []string `json:"tools"`
 	// Group name this workload belongs to
 	Group string `json:"group,omitempty"`
+
+	// Remote server specific fields
+	URL         string             `json:"url,omitempty"`
+	OAuthConfig *RemoteOAuthConfig `json:"oauth_config,omitempty"`
+	Headers     []*registry.Header `json:"headers,omitempty"`
+}
+
+// RemoteOAuthConfig represents OAuth configuration for remote servers
+//
+//	@Description	OAuth configuration for remote server authentication
+type RemoteOAuthConfig struct {
```

Comment:
Could we reuse some of the run config types instead of defining a new structure here?